util/verify_release

```diff
@@ -0,0 +1,36 @@
+#!/usr/bin/env node
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+// A very simple script to verify that a package can be installed and imported.
+// Intended primarily as a basic smoke test after publishing.
+// TODO-RS: Work out how to run this against verdaccio for pre-release verification too!
+
+const { execSync } = require('child_process')
+const path = require('path');
+const tmp = require('tmp')
+var fs = require('fs');
+
+const packageDir = process.argv[2]
+const packageConfigFile = path.join(packageDir, "package.json")
+const packageConfig = JSON.parse(fs.readFileSync(packageConfigFile, 'utf8'))
+const packageName = packageConfig.name
+const packageVersion = packageConfig.version
+const releaseIdentifier = `${packageName}@${packageVersion}`
+
+console.log(`\nVerifying ${releaseIdentifier} release...`)
+
+// Create a temporary directory to act as a dummy consuming project.
+// It's important not to use the current directory because npm will
+// walk the file tree upwards and potentially find unrelated node_modules directories!
+// TODO-RS: An empty directory triggers several warnings, about things like missing
+// package.json and package-lock.json files. Can we clean this up in the future?
+const consumingPackageDir = tmp.dirSync();
+
+// Manually install the dependency
+execSync(`npm install --prefix ${consumingPackageDir.name} ${releaseIdentifier}`)
+
+// Verify that it can be imported. This also ensures that types are available correctly.
+execSync(`ts-node -e "import * as everything from '${packageName}'"`)
```

Comment:
For further improvements:

The package.json has an optional section [files](https://docs.npmjs.com/cli/v6/configuring-npm/package-json#files) that will control what files are actually published. This is generally a good practice. If we add this section to these packages, then we can further enhance this script.

We could create a file in each package `test/verify_publish.ts`. To start each file could just import as you have above. But as we find additional things to test, this is a nice place to put such things. Obviously browser vs node environments might frustrate us, but this is a good start.

Reply:
To be clear, then I think you cp the file and run it in the temp dir...

Reply:
I'm pretty sure I don't like the idea of putting anything that tests the package "from the outside" in the package itself. That always leads to confusion and subtle encapsulation bugs in testing. I think the JS ESDK does the right thing of defining separate `integration-*` packages, and if we want deeper testing we should go down that road instead IMO.